nssp_demo/postprocess_scoring.R

```diff
@@ -0,0 +1,405 @@
+      mae = ae_median,
+      interval_coverage_50,
+      interval_coverage_90
+    )
+
+  summarised_scores <- left_join(abs_wis, rel_wis,
+    by = c("model", "location")
+  )
+  return(summarised_scores)
+}
+
+#' @title Epiweekly Scoring Plot
+#' This function generates a line plot of the weighted interval scores (WIS) for
+#' different models over epiweeks.
+#' @param quantile_scores A scoringutils output table
+#' containing the unsummarised quantile scores.
+#' @param scale A character string specifying the scale
+#' to filter the quantile
+#' scores. Default is "natural".
+#' @return A ggplot object representing the epiweekly scoring plot.
+epiweekly_scoring_plot <- function(quantile_scores, scale = "natural") {
+  epiweekly_score_fig <- quantile_scores |>
+    filter(scale == !!scale) |>
+    mutate(epiweek = epiweek(date), epiyear = epiyear(date)) |>
+    get_pairwise_comparisons(
+      by = c("epiweek", "epiyear"),
+      baseline = "cdc_baseline"
+    ) |>
+    mutate(epidate = forecasttools::epiweek_to_date(
+      epiweek,
+      epiyear,
+      day_of_week = 1
+    )) |>
+    group_by(model, epidate) |>
+    summarise(
+      wis = mean(wis_scaled_relative_skill),
+      .groups = "drop"
+    ) |>
+    as_tibble() |>
+    ggplot(aes(x = epidate, y = wis, color = model)) +
+    geom_line() +
+    geom_point() + # Add points to the line plot
+    labs(
+      title = "Epiweekly Scoring by Model",
+      x = "Epiweek start dates",
+      y = "Relative Weighted Interval Score (WIS)"
+    ) +
+    scale_y_continuous(trans = "log10") +
+    theme_minimal()
+
+  return(epiweekly_score_fig)
+}
+
+location_rel_wis_plot <- function(location, quantile_scores, ...) {
+  return(epiweekly_scoring_plot(
+    quantile_scores |>
+      dplyr::filter(location == !!location),
+    ...
+  ) + ggtitle(
+    glue::glue("Relative WIS over time for {location}")
+  ))
+}
+
+location_score_table <- function(location, quantile_scores, ...) {
+  return
+}
+
+#' Save a list of plots as a PDF, with a
+#' grid of `nrow` by `ncol` plots per page
+#'
+#' @param list_of_plots list of plots to save to PDF
+#' @param save_path path to which to save the plots
+#' @param nrow Number of rows of plots per page
+#' (passed to [gridExtra::marrangeGrob()])
+#' Default `1`.
+#' @param ncol Number of columns of plots per page
+#' (passed to [gridExtra::marrangeGrob()]).
+#' Default `1`.
+#' @param width page width in device units (passed to
+#' [ggplot2::ggsave()]). Default `8.5`.
+#' @param height page height in device units (passed to
+#' [ggplot2::ggsave()]). Default `11`.
+#' @return `TRUE` on success.
+#' @export
+plots_to_pdf <- function(list_of_plots,
+                         save_path,
+                         nrow = 1,
+                         ncol = 1,
+                         width = 8.5,
+                         height = 11) {
+  if (!stringr::str_ends(
+    save_path, ".pdf"
+  )) {
+    cli::cli_abort("Filepath must end with `.pdf`")
+  }
```

Comment:
use [`path_ext`](https://fs.r-lib.org/reference/path_file.html) instead

Reply:
Done in 96f7c77